src/main/java/com/faforever/client/remote/FafServerAccessorImpl.java

```diff
@@ -427,8 +432,22 @@ private void onSessionInitiated(SessionMessage sessionMessage) {
   }
 
   private void logIn(String username, String password) {
-    String uniqueId = uidService.generate(String.valueOf(sessionId.get()), preferencesService.getFafDataDirectory().resolve("uid.log"));
-    writeToServer(new LoginClientMessage(username, Hashing.sha256().hashString(password, UTF_8).toString(), sessionId.get(), uniqueId, localIp));
+    try {
+      String uniqueId = uidService.generate(String.valueOf(sessionId.get()), preferencesService.getFafDataDirectory().resolve("uid.log"));
+      writeToServer(new LoginClientMessage(username, Hashing.sha256().hashString(password, UTF_8).toString(), sessionId.get(), uniqueId, localIp));
+    } catch (IOException e) {
+      onUIDNotFound(e);
+    }
+  }
+
+  @Override
+  protected void onUIDNotFound(Exception e) {
+    if (e.getMessage() == null) {
+      return;
+    }
+    notificationService.addNotification(new ImmediateNotification(i18n.get("UIDNotFound"), e.getMessage(), Severity.ERROR,
+        Collections.singletonList(new ReportAction(i18n, reportingService, e))));
+    logger.error("UID.exe not found", e);
```

Comment:
* It may or may not be called "UID.exe" (and in fact, it's not) so don't assume a file name. Either use the correct name or use an error message like "UID binary could not be found"
* Just because there was an `IOException` doesn't mean "not found"